spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -85,67 +257,181 @@ def write_recording(recording, save_path, acquisition_name='ElectricalSeries', *
+            aux = [isinstance(i, Device) for i in nwbfile.children]
+            if any(aux):
+                device = nwbfile.children[np.where(aux)[0][0]]
+            else:
+                device = nwbfile.create_device(name='Device')
+
+            # Check if 'groups' property exists for channels in Recording
+            if 'group' in recording.get_shared_channel_property_names():
+                el_groups_names = np.unique(recording.get_channel_groups()).tolist()
+            else:
+                el_groups_names = ["0"]
+                # Electrode groups are required for NWB, for consistency we create group for Recording channels
+                ids = recording.get_channel_ids()
+                vals = [0] * len(ids)
+                recording.set_channel_groups(channel_ids=ids, groups=vals)
+            # Creates electrode groups for groups names in Recording
+            for grp_name in el_groups_names:
+                if str(grp_name) not in nwbfile.electrode_groups:
+                    elec_group = nwbfile.create_electrode_group(
+                        name=str(grp_name),
+                        location="electrode_group_location",
+                        device=device,
+                        description="electrode_group_description"
+                    )
+
+            # Check for existing electrodes
+            if nwbfile.electrodes is not None:
+                nwb_elec_ids = nwbfile.electrodes.id.data[:]
+            else:
+                nwb_elec_ids = []
+
+            # add new electrodes with id, (x, y, z) and groups
+            for m in channel_ids:
+                if m not in nwb_elec_ids:
+                    location = recording.get_channel_property(m, 'location')
+                    while len(location) < 3:
+                        location = np.append(location, [0])
+                    impedence = -1.0
+                    grp_name = recording.get_channel_groups(channel_ids=[m])
+                    grp = nwbfile.electrode_groups[str(grp_name[0])]
+                    nwbfile.add_electrode(
+                        id=m,
+                        x=float(location[0]), y=float(location[1]), z=float(location[2]),
+                        imp=impedence,
+                        location='unknown',
+                        filtering='none',
+                        group=grp,
+                    )
+            electrode_table = nwbfile.electrodes
+
+            # add/update electrode properties
+            for ch in channel_ids:
+                rx_channel_properties = recording.get_channel_property_names(channel_id=ch)
+                for pr in rx_channel_properties:
+                    val = recording.get_channel_property(ch, pr)
+                    desc = 'no description'
+                    # property 'location' of RX channels corresponds to x, y, z of NWB electrodes
+                    if pr == 'location':
+                        names = ['x', 'y', 'z']
```

Comment:
The 3d location is cool, but currently many sorters only support 2d locations. Would it be possible to only load x, y as locations? otherwise we would need to take care of the 3d locations and refractor many sorters.

You can check how I did it in the [MEArecRecordingExtractor](https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/mearecextractors/mearecextractors.py#L51-L63)